src/pages/SubmissionDetails.js

```diff
@@ -128,7 +128,9 @@ export const SubmissionDetails = ({ submissionId }) => {
       }
     }
 
-    loadRunSetData().then(runSet => runSet && loadMethodsData(runSet.method_version_id))
+    setLoading(true)
```

Comment:
I think it would be better to use [withBusyState](https://github.com/DataBiosphere/cbas-ui/blob/8798c52740ddca0dbf0715661bb0ecdbc24cea1f/src/libs/utils.js#L135-L142) instead of manually setting the flag. Once it is set to true only refreshing the component will set it back to false. `withBusyState` will do that for you - set the flag to true, wait for the passed function to finish and then set the flag back to false. 

Link to a usage example in [JobHistory page in Terra UI](https://github.com/DataBiosphere/terra-ui/blob/f7964e84f2d1ef3d79ac5ad842cf2deb5685043d/src/pages/workspaces/workspace/JobHistory.js#L137).

Reply:
this looks cool - I can try this